app/code/Magento/Captcha/Observer/CheckUserForgotPasswordBackendObserver.php

```diff
@@ -89,4 +106,19 @@ public function execute(\Magento\Framework\Event\Observer $observer)
 
         return $this;
     }
+
+    /**
+     * Return the shared request.
+     * If the request wasn't injected because of the backward compatible optional constructor dependency,
+     * create a new request instance.
+     *
+     * @return RequestInterface
+     */
+    private function getRequest(): RequestInterface
+    {
+        if (null === $this->request) {
```

Comment:
Why not to move it inside constructor?

Reply:
Lazy loading of the class.
Actually - there's no huge performance difference.